src/Processors/Formats/Impl/ArrowFieldIndexUtil.h

```diff
@@ -0,0 +1,45 @@
+#pragma once
+#include "config.h"
+#if USE_PARQUET || USE_ORC
+#include <map>
+#include <Core/Block.h>
+#include "DataTypes/Serializations/ISerialization.h"
+namespace arrow
+{
+class Schema;
+class DataType;
+class Field;
+}
+namespace DB
+{
+class ArrowFieldIndexUtil
+{
+public:
+    /// For orc format, nested_type_has_index_ = true.
+    explicit ArrowFieldIndexUtil(bool ignore_case_, bool nested_type_has_index_, bool allow_missing_columns_)
+        : ignore_case(ignore_case_)
+        , nested_type_has_index(nested_type_has_index_)
+        , allow_missing_columns(allow_missing_columns_)
+    {
+    }
```

Comment:
I think instead of `nested_type_has_index` argument class can be templatized with `template <bool idx>` and then use `idx` everywhere directly as integer (`false` is 0 and `true` is 1) instead of conditional expressions...
at your discretion though - don't think it makes better performance